### PR TITLE
[js/webgpu] Remove the limitation on axis in softmax

### PR DESCRIPTION
**Pull request by gyagp**
In current implementation, axis in softmax has to be the last, which is an obvious limitation. This PR removes this limitation and will fix issues #20710 and #22176.

### CODE REVIEW

**Comment by gyagp** [comment]
run web CI

File: js/web/lib/wasm/jsep/webgpu/ops/softmax.ts [hunk 1]
```diff
@@ -30,19 +31,35 @@ export interface SoftmaxAttributes extends AttributeWithCacheKey {
...
+
+  if (needTranspose) {
+    const perm = [];
+    for (let i = 0; i < axis; i++) {
+      perm.push(i);
+    }
+    perm.push(inputRank - 1);
+    for (let i = axis + 1; i < inputRank; i++) {
+      perm.push(i - 1);
+    }
```
Comment (qjia7) [inline js/web/lib/wasm/jsep/webgpu/ops/softmax.ts:52]: It seems not correct. I would expect axis is the last dim in perm. But it's not. For example , the original dims index array is [0, 1, 2, 3, 4, 5] while axis = 2. The expected perm will be [0, 1, 5, 3, 4, 2]. But actually get [0, 1, 5, 2, 3, 4].

Comment (gyagp) [reply js/web/lib/wasm/jsep/webgpu/ops/softmax.ts]: My previous understanding on perm is wrong. I just uploaded another PR to fix it. Thanks!

**Review by qjia7 — APPROVED** [review]
LGTM, thanks.

**Review by fs-eire — APPROVED** [review]

**Pull request merged**